loader/loader.go

```diff
@@ -547,11 +549,41 @@ func (l *Loader) closeFileJobQueue() {
 	l.fileJobQueueClosed.Set(true)
 }
 
+// align with https://github.com/pingcap/dumpling/pull/140
+// if input is malformed, return original string and print log
+func unescapePercent(input string, logger log.Logger) string {
+	buf := bytes.Buffer{}
+	buf.Grow(len(input))
+	i := 0
+	for i < len(input) {
+		if input[i] != '%' {
+			buf.WriteByte(input[i])
+			i++
+		} else {
+			if i+2 >= len(input) {
+				logger.Error("malformed filename while unescapePercent", zap.String("filename", input))
+				return input
+			}
+			ascii, err := hex.DecodeString(input[i+1 : i+3])
+			if err != nil {
+				logger.Error("malformed filename while unescapePercent", zap.String("filename", input))
+				return input
+			}
+			buf.WriteString(fmt.Sprintf("%s", ascii))
```

Comment:
Can we use `buf.Write` directly here?

Reply:
https://github.com/pingcap/dm/pull/980/commits/1d7c8df0599191e738b3cdb709a27a913df82586